### PR TITLE
dts: arm: nordic: fix cryptocell description

### PR DESCRIPTION
**Pull request by gmarull**
The ARM Cryptocell 310/312 IP is wrapped by Nordic specific registers. It is organized as follows:

- Base address: Nordic wrapper
- Base address + 0x1000: ARM Cryptocell IP registers

Following more standard devicetree conventions, use a single node for what is exposed as a single peripheral. The node contains 2 register entries, one for the wrapper and a second one for the 3rd party IP. Compatibles are used from more specific (nordic,cryptocell) to more generic (arm,cryptocell-3xx).

Other minor fixes: peripheral is disabled by default (as it should be in SoC dts files).

### CODE REVIEW

File: dts/arm/nordic/nrf52840.dtsi [hunk 1]
```diff
@@ -537,16 +537,11 @@
...
-			cryptocell310: crypto@5002b000 {
-				compatible = "arm,cryptocell-310";
-				reg = <0x5002B000 0x1000>;
-				interrupts = <42 NRF_DEFAULT_IRQ_PRIORITY>;
-			};
+			compatible = "nordic,cryptocell", "arm,cryptocell-310";
+			reg = <0x5002a000 0x1000>, <0x5002b000 0x1000>;
+			reg-names = "wrapper", "core";
+			interrupts = <42 NRF_DEFAULT_IRQ_PRIORITY>;
+			status = "disabled";
```
Comment (57300) [inline dts/arm/nordic/nrf52840.dtsi:544]: This lgtm, but I would suggest having just one compatible here, like before.

Having multiple compatibles influences driver selection. If there existed a generic driver for `arm,cryptocell-310`, then I'd assume it should be able to drive this peripheral using the non-Nordic set of registers. That would probably require putting the "core" registers aside into a separate node. If that is not possible, then this peripheral should only be Nordic compatible.

This is purely theoretical, and I'm not familiar with CryptoCell, so I don't know if that makes sense.

**Review by 57300 — CHANGES_REQUESTED** [review]
Replacing the `nordic,nrf-cc3xx` bindings will require changes to this file:
https://github.com/zephyrproject-rtos/zephyr/blob/c82d0232485aa75525298985cf805c569620e6e9/soc/arm/nordic_nrf/Kconfig.peripherals#L15-L19

But before that, please consider my comment below.

File: dts/arm/nordic/nrf52840.dtsi (hunk 1 above)
Comment (gmarull) [reply dts/arm/nordic/nrf52840.dtsi:544]: having one compatible doesn't follow devicetree spec, IMHO. 

> The compatible property value consists of one or more strings that define the specific programming model for
the device. This list of strings should be used by a client program for device driver selection. The property
value consists of a concatenated list of null terminated strings, from most specific to most general. They allow
a device to express its compatibility with a family of similar devices, potentially allowing a single device driver
to match against several devices.

The only specific bit of this IP integration is the extra Nordic wrapper used to enable it. So in theory, a single vendor-agnostic driver could be implemented, provided it supports vendor quirks (wrapper in this case).

Comment (57300) [reply dts/arm/nordic/nrf52840.dtsi:544]: > a single vendor-agnostic driver could be implemented, provided it supports vendor quirks (wrapper in this case).

That's what I was unsure about, but if this is acceptable then I'm OK with it :+1:

**Review by 57300 — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**